alternator/executor.cc

```diff
@@ -2951,11 +2951,38 @@ static future<executor::request_return_type> do_query(schema_ptr schema,
     });
 }
 
+static dht::token token_for_segment(int segment, int total_segments) {
+    assert(total_segments > 1 && segment < total_segments);
+    int64_t delta = std::numeric_limits<int64_t>::max() / total_segments * 2;
+    return dht::token::from_int64(std::numeric_limits<int64_t>::min() + 1 + delta * segment);
```

Comment:
I slept under a rock, and suddenly tokens are *always* 64 bits?

Reply:
And where does the "+1" come from? Is min() itself considered an illegal token? What token comes after max()? Where did you find this documented?

Reply:
I just noticed below you never call token_for_segment() for segment==0, so this + 1 doesn't even seem to make a difference...

Reply:
Indeed, + 1 doesn't make sense anymore with the special segment==0 code, it's a good idea and I'll drop it in v2. 

As for the other questions:
 * yes, we successfully deprecated all non-conforming partitioners, and token is now guaranteed to be 64 bits, which will allow keeping it per table and serializing/deserializing it without the knowledge of its partitioner - since it will always be representable as int64 underneath. Browse the mailing list with `partitioner` keyword and you'll see several related patches by PiotrJ and others.
 * +1 came from the fact that I glimpsed in another series on the mailing list that min() is a special case. I can't find it now, but it was one of PiotrJ's patches about using a correct int distribution when choosing a random token.